src/taipy/core/data/_data_manager.py

```diff
@@ -101,35 +104,46 @@ def __create(
             raise InvalidDataNodeType(data_node_config.storage_type)
 
     @classmethod
-    def _clean_pickle_file(cls, data_node: Union[DataNode, DataNodeId]):
-        data_node = cls._get(data_node) if isinstance(data_node, str) else data_node
+    def _clean_pickle_file(cls, data_node: DataNode):
         if not isinstance(data_node, PickleDataNode):
             return
         if data_node.is_generated and os.path.exists(data_node.path):
             os.remove(data_node.path)
 
     @classmethod
-    def _clean_pickle_files(cls, data_nodes: Iterable[Union[DataNode, DataNodeId]]):
+    def _update_preserve_files(cls, data_node: DataNode):
+        if isinstance(data_node, _AbstractFileDataNode):
+            _AbstractFileDataNode._check_and_update_preserve_file(old_path=data_node.path)
+
+    @classmethod
+    def _update_preserve_file_and_clean_pickle_file(cls, data_node: Union[DataNode, DataNodeId]):
+        data_node = cls._get(data_node) if isinstance(data_node, str) else data_node
+        cls._clean_pickle_file(data_node)
+        cls._update_preserve_files(data_node)
+
+    @classmethod
+    def _update_preserve_files_and_clean_pickle_files(cls, data_nodes: Iterable[Union[DataNode, DataNodeId]]):
```

Comment:
I would not put together the preserve file update and the pickle cleaning. These are completely unrelated features. I would split them in the code as well.